### PR TITLE
Rename target to local_alias in AsName

### PR DESCRIPTION
**Pull request by smazouz42**
This change updates the terminology used in `AsName` from "target" to "local_alias" to better represent the name's function within the local context. This enhancement improves clarity and consistency in the codebase.
Fixes issue #1929

### CODE REVIEW

File: pyccel/parser/semantic.py [hunk 1]
```diff
@@ -4320,8 +4320,8 @@ def _insert_obj(location, target, obj):
             import_init = p.semantic_parser.ast.init_func if source_target not in container['imports'] else None
             import_free = p.semantic_parser.ast.free_func if source_target not in container['imports'] else None
             if expr.target:
-                targets = {i.target if isinstance(i,AsName) else i:None for i in expr.target}
-                names = [i.name if isinstance(i,AsName) else i for i in expr.target]
+                targets = {i.target if isinstance(i,AsName) else i:None for i in expr.local_alias}
+                names = [i.name if isinstance(i,AsName) else i for i in expr.local_alias]
```
Comment (EmilyBourne) [inline pyccel/parser/semantic.py:4324]:
```suggestion
                targets = {i.local_alias if isinstance(i,AsName) else i:None for i in expr.target}
                names = [i.name if isinstance(i,AsName) else i for i in expr.target]
```

**Comment by EmilyBourne** [comment]
@smazouz42 do you plan to complete this PR or should we pick it up and finish it?

**Review by jalalium — APPROVED** [review]

**Pull request merged**